src/rpc/mining.cpp

```diff
@@ -302,7 +302,7 @@ static UniValue getblocktemplate(const JSONRPCRequest& request)
                 "    https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki#getblocktemplate_changes\n"
                 "    https://github.com/bitcoin/bips/blob/master/bip-0145.mediawiki\n",
                 {
-                    {"template_request", RPCArg::Type::OBJ, RPCArg::Optional::NO, "A json object in the following spec",
+                    {"template_request", RPCArg::Type::OBJ, "{\"mode\":\"template\"}", "A json object in the following spec",
```

Comment:
Does this need backport? Should be a separate pull request, since behavior changes should not be mixed with refactoring.

Reply:
It may be wrong doc-wise, but tests break if you make this mandatory. And it *is* optional in the code.

Reply:
Which tests break?

Reply:
minig_basic.py (see above log)

Reply:
Ah, so the default is not `{\"mode\":\"template\"}`, but the empty dict `{}`? The help text for mode says "This must be set to \"template\", \"proposal\" (see BIP 23), **or omitted**" (emphasis mine)

Reply:
If it can be omitted, `Optional::NO` is probably wrong, then. I can change the default to simply say `{}`, but the code essentially acts as if it was `{"mode":"template"}`..